src/jupyterlabcontroller/services/events.py

```diff
@@ -0,0 +1,34 @@
+import asyncio
+from collections.abc import AsyncGenerator
+from typing import Optional
+
+from sse_starlette.sse import ServerSentEvent
+from structlog.stdlib import BoundLogger
+
+from ..models.v1.external.event import EventMap, EventQueue
+
+
+class EventManager:
+    def __init__(self, logger: BoundLogger, events: EventMap) -> None:
+        self.logger = logger
+        self.events = events
+
+    async def user_event_publisher(
+        self,
+        username: str,
+    ) -> AsyncGenerator[ServerSentEvent, None]:
+        try:
+            while True:
+                evs: Optional[EventQueue] = self.events.get(username)
+                if evs:
+                    for ev in evs:
```

Comment:
I did this a bit differently--mostly, I (at least at this stage) want to raise an exception if I ask for an event queue for a user that doesn't have one, since I already *should* have one.  I'm not sure the events object itself ever does become a real object (assuming we have a good way of serializing a deque for storage in Redis, anyway), but the event map (which holds the user-to-queue mapping) is now a real object.  And the test there is gone, I just loop through the events.